fossunited/payments/doctype/razorpay_payment/test_razorpay_payment.py

```diff
@@ -0,0 +1,137 @@
+
+        # Have taken this metadata to mimic the razorpay payment metadata
+        meta_data = {
+            "attendees": [
+                {
+                    "designation": fake.job(),
+                    "email": fake.email(),
+                    "full_name": fake.name(),
+                    "organization": fake.company(),
+                    "placeholder": fake.name(),
+                    "wants_tshirt": 0,
+                },
+                {
+                    "designation": fake.job(),
+                    "email": fake.email(),
+                    "full_name": fake.name(),
+                    "organization": fake.company(),
+                    "placeholder": fake.name(),
+                    "wants_tshirt": 0,
+                },
+            ],
+            "custom_fields": {
+                "food_preference": fake.random_element(elements=("Veg", "Non-Veg", "Vegan"))
+            },
+            "event": self.event,
+            "num_seats": "2",
+            "tier": {
+                "creation": str(tier.creation),
+                "currency": tier.currency,
+                "description": tier.description,
+                "docstatus": tier.docstatus,
+                "doctype": "FOSS Ticket Tier",
+                "enabled": tier.enabled,
+                "idx": tier.idx,
+                "maximum_tickets": tier.maximum_tickets,
+                "modified": str(tier.modified),
+                "modified_by": tier.modified_by,
+                "name": tier.name,
+                "owner": tier.owner,
+                "parent": tier.parent,
+                "parentfield": "tiers",
+                "parenttype": "FOSS Chapter Event",
+                "price": tier.price,
+                "title": tier.title,
+                "valid_till": str(tier.valid_till),
+            },
+        }
+
+        invoice = frappe.get_doc(
+            {
+                "doctype": "Razorpay Payment",
+                "document_type": EVENT,
+                "document_name": self.event,
+                "email": user.email,
+                "amount": tier.price * 2,  # 2 attendees
+                "currency": "INR",
+                "status": "Pending",
+                "meta_data": json.dumps(meta_data),
+            }
+        )
+
+        try:
+            invoice.insert()
+            frappe.db.commit()
+            return invoice
+        except frappe.PermissionError:
+            frappe.db.rollback()
+            return None
+
+    def test_admin_can_create_invoice(self):
+        invoice = self.create_payment_document(self.admin_user)
+        self.assertIsNotNone(invoice)
+        if invoice:
+            self.assertEqual(invoice.doctype, RAZORPAY_PAYMENT)
+            self.assertEqual(invoice.owner, self.admin_user.name)
+
+    def test_normal_user_cannot_create_invoice(self):
+        invoice = self.create_payment_document(self.normal_user)
+        self.assertIsNone(invoice)
```

Comment:
non-actionable comment - another way to write the test would be to explicitly catch the `frappe.PermissionError`. the system will raise a `PermissionError` if this situation somehow happens so the test should emulate the system behavior.

```suggestion
        with self.assertRaises(frappe.PermissionError):
            invoice = self.create_payment_document(self.normal_user)
```

+ the necessary change in `self.create_payment_document` that re-raises the exception after performing the rollback